src/test2.py

```diff
@@ -190,27 +190,28 @@ def mul_vars_string(var_list):
     if var_list == []:
         return const_string(1)
     elif len(var_list) == 1:
-        return var_string(var_list[0])
-    return "(poly.mul " + var_string(var_list[0]) + " " + mul_vars_string(var_list[1:]) + ")"
+        return var_list[0]
```

Comment:
Here and below, I had to make some changes to the printing output to handle powers. I think this part of the code could be simplified/reengineered.